src/ui/library/Skeleton.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {View, StyleSheet, Animated, Easing} from 'react-native';
+import LinearGradient from 'react-native-linear-gradient';
+
+type Props = {
+  width: number;
+  height?: number;
+  circle?: boolean;
+};
+
+const AnimatedLG = Animated.createAnimatedComponent(LinearGradient);
+
+export function Skeleton({width, height, circle = false}: Props) {
+  const widthValue = !circle ? `${width}%` : width;
+  const heightValue = height ? height : !circle && !height ? 25 : width;
+  const borderRadius = circle ? Math.floor(width / 2) : 4;
+
+  const animatedValue = new Animated.Value(0);
+  React.useEffect(() => {
+    Animated.loop(
+      Animated.timing(animatedValue, {
+        toValue: 1,
+        duration: 2000,
+        easing: Easing.linear,
+        useNativeDriver: true,
+      }),
+    ).start();
+  });
+
+  const translateX = animatedValue.interpolate({
+    inputRange: [0, 1],
+    outputRange: [circle ? -width : -250, circle ? width : 250],
```

Comment:
You could also calculate the output with the width or screen dimension so it can fit good in bigger screens as well if the loading state placeholder should take more horizontal space

Reply:
Since the width becomes a percentage in case of a rectangular skeleton we need to calculate the actual width onLayout and this makes the animation a bit shaky. I did implemented the screen width calculation and removed it. Will put it back.